docs/specifications/tests/Connectivity-TP/connectivity03.md

```diff
@@ -1,67 +1,214 @@
-4. To obtain the AS numbers for IPv4 addresses; first reverse the four numerical
-octets and place dots between each.  
-4.1. At the end of the reversed IPv4 address concatenate the string (from step 2)  
-4.2 Send a "TXT" query using the string (obtained from Step 4.1)  
-4.3 If there is an ANSWER, go to step 4.4, else go to the next string
-in the list (from step 2)  
-4.4 The AS number for the IPv4 address is found in the ANSWER for the query 
-5. If all the retrieved AS (obtained from step 3.3 and 4.3) are same, then the test
-   fails.
 
-### Outcome(s)
+## Ordered description of steps to be taken to execute the test case
 
-If there is a AS which is different from the other retrieved AS, then the
-test succeeds.
+1. Obtain the total set of IP addresses of the name servers for the 
+   *child zone* using [Method4] and [Method5].
 
-### Special procedural requirements
+2. For each IP address in the set from step 1 above, determine the ASN set
+   announcing the IP address using either the *[Cymru database]* or the 
+   *[RIPE database]* as described in separate sections below. 
 
-None
+3. For each IP protocol (IPv4, IPv6) do:
+   1. If all addresses are announced from one and the same ASN emit
+      *[IPV4_ONLY_ONE_ASN]* or *[IPV6_ONLY_ONE_ASN]*.
+   2. If all addresses are announced from the same ASN set of multiple 
+      ASNs, emit *[IPV4_SAME_ASN]* or *[IPV6_SAME_ASN]*.
+   3. If the addresses are announced from overlapping ASN sets, i.e. 
+      every set is either a subset, identical or superset of all other sets, 
+      but there are at least two IP addresses with non-identical ASN sets,
+      emit *[IPV4_OVERLAP_ASN]* or *[IPV6_OVERLAP_ASN]*.
+   4. If there are at least two IP addresses for which the ASN sets are 
+      neither a subset, identical nor superset of the other emit
+      *[IPV4_DIFFERENT_ASN]* or *[IPV6_DIFFERENT_ASN]*.
+
+
+## Outcome(s)
+
+The outcome of this Test Case is "fail" if there is at least one 
+message with the severity level ERROR or CRITICAL.
+
+The outcome of this Test Case is "warning" if there is at least 
+one message with the severity level WARNING, but no message with 
+severity level ERROR or CRITICAL.
+
+In other cases the outcome of this Test Case is "pass".
+
+Message            |Default severity level (if message is emitted)
+:------------------|:------------
+EMPTY_ASN_SET      |ERROR        
+ERROR_ASN_DATABASE |ERROR        
+IPV4_ONLY_ONE_ASN  |ERROR        
+IPV4_SAME_ASN      |NOTICE       
+IPV4_OVERLAP_ASN   |NOTICE       
+IPV4_DIFFERENT_ASN |INFO         
+IPV6_ONLY_ONE_ASN  |ERROR        
+IPV6_SAME_ASN      |NOTICE       
+IPV6_OVERLAP_ASN   |NOTICE       
+IPV6_DIFFERENT_ASN |INFO         
+
+
+## Special procedural requirements
+
+This test case is dependent on one of two possible services that can provide
+ASN lookup, RIPE or Cymru. The service must be available over the network.
+
+
+### Cymru ASN lookup
```

Comment:
The steps here are too detailed. Instead of taking the responsibility to explain many of the intricacies, this instruction should defer as much as possible to the Cymru documentation. If it's too much to edit that right now, an issue should be created as a reminder. Same goes for the RIPE procedure.

Reply:
If it is OK I rather keep for now.